test/evp_test.c

```diff
@@ -2214,28 +2214,28 @@ static int encapsulate(EVP_TEST *t, EVP_PKEY_CTX *ctx, const char *op,
     KEM_DATA *kdata = t->data;
     unsigned char *wrapped = NULL, *secret = NULL;
     size_t wrappedlen = 0, secretlen = 0;
-    OSSL_PARAM params[10] = { OSSL_PARAM_END };
+    OSSL_PARAM params[10];
     size_t params_n = 0;
 
+    memset(params, 0, sizeof(params));
+
     if (sk_OPENSSL_STRING_num(kdata->init_ctrls) > 0)
         if (ctrl2params(t, kdata->init_ctrls, NULL, params,
-                        OSSL_NELEM(params) - 1, &params_n))
+                        OSSL_NELEM(params) - 2, &params_n))
             goto err;
 
     /* We don't expect very many controls here */
-    if (params_n >= 9) {
-        TEST_error("Too many encapsulate test case parameters %zd", params_n);
+    if (!TEST_size_t_lt(params_n, OSSL_NELEM(params) - 2)) {
+        TEST_info("Too many encapsulate test case parameters %zd", params_n);
         goto err;
     }
 
-    if (kdata->entropy != NULL) {
+    if (kdata->entropy != NULL)
         /* Input key material a.k.a entropy */
-        params[params_n] =
+        params[params_n++] =
             OSSL_PARAM_construct_octet_string(OSSL_KEM_PARAM_IKME,
                                               kdata->entropy,
                                               kdata->entropylen);
-        params[params_n + 1] = OSSL_PARAM_construct_end();
```

Comment:
I would feel more comfortable with the proper version of this always outside this conditional to properly terminate `params`.